core/src/com/unciv/logic/city/City.kt

```diff
@@ -91,7 +92,15 @@ class City : IsPartOfGameInfoSerialization, INamed {
     var hasSoldBuildingThisTurn = false
     var isPuppet = false
     var shouldReassignPopulation = false  // flag so that on startTurn() we reassign population
-
+    
+    // HashMap to check if we can build the unitType with the saved promotion. 
+    // (Saved promotion: meaning that the player checkbox to save the promotion when adding them to a unitType.)
+    // To not have to redo it for every unit build afterword.
+    var canBuildUnitTypeWithSavedPromotion = HashMap<String, Boolean>()
+    
+    // To stroe unitType to promotion.
```

Comment:
Remove comment, call this "unitTypeToPromotion"
We already know it's for city since it's in the city class